### DISCUSSION TITLE
Refresh Table on new Column definition

### DISCUSSION THREAD

**Discussion by Thoemmy** [revolist/revogrid#614] [General]
I have a question regarding refreshing the column headers after the json hass been modified dynamically with user input.

How can i refresh the column headers in Angular properly without making a page reload?

Inside revogrid the columns got updated if i change the columns and transfer it into the revogrid attribute columns.
But currently ony the width of the rows are adjusted without the column header, this only gots refreshed after page reload or if you manually resize the column header.

The refresh('all') method is undefined in angular but i guess this is only for rgRows as written in the docs.

Would be nice if ou could give me a suggestion.

**Comment by Thoemmy** [comment]
Also the method updateColumns is not defined:
```
ERROR TypeError: Cannot read properties of undefined (reading 'updateColumns')
    at TableComponent.ngOnChanges (table.component.ts:72:21)
```

**Comment by MattTheCuber** [comment]
I don't know Angular very well, but couldn't you bind the columns to a dynamic state variable so that the columns update when you change the data in that state variable? That is how we do it in Vue.

**Reply by Thoemmy** [reply]
Yes, I do that with the source and with the columns (ngrx), the thing is that if I change the name of the column, then the name is updated immediately, but if I adjust the column width then this is only applied to the columns but not for the column header.

I hope this explaination delivers more clarity 🙈

Currently the only way i found (not the best one) is to reload the page after saving the table settings.

**Comment by revolist** [comment]
Hi @Thoemmy, I didn't understand your case scenario. What @MattTheCuber suggested it makes sense, you need to trigger viewport refresh for the column header to be updated. To do so you can just put new columns ref or you can make a plugin which updates column data source store `items` collection. This would trigger update as well but more advanced way.

**Reply by Thoemmy** [reply]
The issue is that, when the column definition is updated, all settings are applied successfully, and the column body adjusts in width, but the column header does not reflect the updated width.

So i am changing the column ref, also the change detection of Angular is triggered but the column header dont change.

**Reply by revolist** [reply]
@Thoemmy Could you make a sample with 1 button which is demonstrating the issue? 

From your description it sounds as an issue

**Reply by Thoemmy** [reply]
@revolist i will first try the newest release and if its still remain i will setup a simple example and share it here.

Thats why i initially did not create an issue and started a discussion.

**Reply by revolist** [reply]
Sure! Thanks!

**Reply by Thoemmy** [reply]
@revolist i updated the package to the newest version and the issue is solved.

**Reply by m2a2x** [reply]
Thanks for the update!
